src/js/components/HealthChecksComponent.jsx

```diff
@@ -184,10 +199,39 @@ var HealthChecksComponent = React.createClass({
                 }
                 fieldId={`${fieldsetId}.${i}.portIndex`}
                 label="Port Index"
-                value={row.portIndex}>
+                value={row.portIndex || 0}>
                 <input ref={`portIndex${i}`} {...numberInputAttributes} />
               </FormGroupComponent>
             </div>
+            <div className={portNumberClassSet}>
+              <FormGroupComponent
+                errorMessage={
+                  getErrorMessage(`${fieldsetId}.${i}.port`)
+                }
+                fieldId={`${fieldsetId}.${i}.port`}
+                label="Port Number"
+                value={row.port || 0}>
```

Comment:
@pierlo-upitup I think we do it right already. It's not directly optional, but this:
```For TCP/HTTP health checks, you need to specify either port or portIndex```
So we cover this behavior in this PR. Everything fine.